roles/pulp_api/tasks/generate_token_auth_key.yml

```diff
@@ -0,0 +1,17 @@
+---
+- name: Look for token authentication key
+  stat:
+    path: "{{ __pulp_common_pulp_pki_dir }}/private.pem"
+    get_attributes: false
+    get_checksum: false
+    get_mime: false
+  register: __pulp_token_auth_key
+
+- name: Generate token authentication private key
+  openssl_privatekey:
+    path: "{{ __pulp_common_pulp_pki_dir }}/private.pem"
+    type: ECC
+    curve: secp256r1
+    owner: "{{ pulp_user }}"
+    group: "{{ pulp_group }}"
+  when: not __pulp_token_auth_key.stat.exists
```

Comment:
You need become, so this does not simply run as ansible_user .

This is definitely true for the 2nd task, not sure about the 1st.

Reply:
I do not know either. But become with pulp_user should be ok for both.